app/controllers/kaui/payment_methods_controller.rb

```diff
@@ -75,6 +75,30 @@ def restful_show
     redirect_to kaui_engine.account_path(payment_method.account_id)
   end
 
+  def validate_external_key
+    external_key = params.require(:external_key)
+
+    begin
+      payment_methods = Kaui::PaymentMethod::find_in_batches_by_search_key(external_key, 0, 100, options_for_klient)
```

Comment:
While this works, it is quite expensive (it executes a `LIKE` query on various columns).

We should instead use [this API](https://github.com/killbill/killbill/blob/4b09131907ba8f111997482bb8fa93ebd31c9583/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentMethodResource.java#L119-L138) (it doesn't look like it's supported by the Ruby client yet, but should be easy to add).